src/services/services.ts

```diff
@@ -3282,7 +3282,9 @@ export function getPropertySymbolsFromContextualType(node: ObjectLiteralElementW
         // Bad discriminant -- do again without discriminating
         return mapDefined(contextualType.types, t => t.getProperty(name));
     }
-    return discriminatedPropertySymbols;
+    // by eliminating duplicates we might even end up with a single symbol
+    // that helps with displaying better quick infos on properties of union types
+    return deduplicate(discriminatedPropertySymbols, equateValues);

```

Comment:
I don't understand why this "works" -- which symbol gets discarded, and how do we know the one that's being kept is the one with doc comments attached?

Reply:
They're all the same symbol, so wouldn't all of that info be the same? It's just that we're q	querying potentially the same info from multiple angles and don't dedupe?

(Why don't we do this in the above mapDefined case, too?)

Reply:
It seems like the deduping should be happening upstream of this, or that deduping should already be a prevalent pattern in this file - surely this isn't the only place where we might end up with multiple instances of the same symbol in the list?

Reply:
This works because only a single symbol is allowed by the caller here:
https://github.com/microsoft/TypeScript/blob/b4787652d281188da1d61fe9cc3a17589f20c9fc/src/services/services.ts#L3255-L3258

We can't dedupe this too soon because it would interfere with `.length`-based checks like this one:
https://github.com/microsoft/TypeScript/blob/b4787652d281188da1d61fe9cc3a17589f20c9fc/src/services/services.ts#L3277

I agree though that perhaps this deduping should also happen here:
https://github.com/microsoft/TypeScript/blob/b4787652d281188da1d61fe9cc3a17589f20c9fc/src/services/services.ts#L3283

but it would be cool to figure out a test case for it before making the change.